src/engine/sim_engine/filesystem.rs

```diff
@@ -29,8 +28,8 @@ impl SimFilesystem {
 }
 
 impl Filesystem for SimFilesystem {
-    fn devnode(&self) -> EngineResult<PathBuf> {
-        Ok(PathBuf::from(format!("/dev/stratis/{}", self.name)))
+    fn devnode(&self) -> PathBuf {
+        vec!["/dev/stratis", &self.name].into_iter().collect()
```

Comment:
No need for vec allocation. You should be able to just do

```rust
["/dev/stratis",&self.name].iter().collect()
```

Reply:
Done, thanks!